frontend/appflowy_flutter/lib/ai/widgets/prompt_input/desktop_prompt_input.dart

```diff
@@ -221,6 +223,26 @@ class _DesktopPromptInputState extends State<DesktopPromptInput> {
     );
   }
 
+  void checkForAskingAI() {
+    final paletteBloc = context.read<CommandPaletteBloc?>(),
+        paletteState = paletteBloc?.state;
+    final isAskingAI = paletteState?.askAI ?? false;
+    if (!isAskingAI) return;
+    final query = paletteState?.query ?? '';
+    if (query.isEmpty) return;
+    final sources =
+        (paletteState?.askAISources ?? []).map((e) => e.id).toList();
+    final metadata =
+        context.read<AIPromptInputBloc?>()?.consumeMetadata() ?? {};
+    final promptState = context.read<AIPromptInputBloc?>()?.state;
+    final predefinedFormat = promptState?.predefinedFormat;
+    if (sources.isNotEmpty) {
+      widget.onUpdateSelectedSources(sources);
+    }
+    widget.onSubmitted.call(query, predefinedFormat, metadata);
+    paletteBloc?.add(CommandPaletteEvent.askedAI());
+  }
```

Comment:
We can return the null values easier to avoid using too many `?`.


```suggestion
  void checkForAskingAI() {
    final paletteBloc = context.read<CommandPaletteBloc?>();
    final paletteState = paletteBloc?.state;

    if (paletteBloc == null || paletteState == null) {
      return;
    }

    if (!paletteState.askAI) return;

    final query = paletteState.query;
    if (query == null || query.isEmpty) return;

    final sources = paletteState.askAISources
        ?.where((source) => source.selected)
        .map((e) => e.id)
        .toList();
    
    if (sources != null && sources.isNotEmpty) {
      widget.onUpdateSelectedSources(sources);
    }

    final promptBloc = context.read<AIPromptInputBloc?>();
    if (promptBloc == null) {
      return;
    }

    final metadata = promptBloc.consumeMetadata();
    final predefinedFormat = promptBloc.state.predefinedFormat;

    widget.onSubmitted(query, predefinedFormat, metadata);
    paletteBloc.add(const CommandPaletteEvent.askedAI());
  }
```

Reply:
fixed, some of the logic cannot be returned directly